### PR TITLE
Faster runtime of predict_win and predict_draw

### PR DESCRIPTION
**Pull request by philihp**
## Description of Changes

* [x] Adds some unit tests that I have for the Javascript library, to prevent any regressions.
* [x] Tells `itertools.permutations` to give permutation pairs, rather than all full permutations
* [x] Additionally, the length of permutation_pairs is shorter, and that simplifies the denominator to just a triangular number.

I noticed that for teams of ABCD, your code as written was finding all permutations (ABCD, ABDC, ACBD, ACDB, ...) and then only using the first two from the permutation, which for n>=4 teams, this causes the same pairings to be calculated multiple times.

This should reduce runtime from O(n^n) to O(n^2).

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under openskill.py's MIT license.

I certify the above statement is true and correct:
Philihp Busby

### CODE REVIEW

File: openskill/rate.py [hunk 1]
```diff
@@ -216,22 +217,10 @@ def predict_win(teams: List[List[Rating]], **options) -> List[Union[int, float]]
...
-            aggregate_sum = sum(aggregate)
-            aggregate_multiple = n
-            for length in range(1, n - 2):
-                aggregate_multiple *= n - length
-            probabilities.append(aggregate_sum / aggregate_multiple)
-        return probabilities
-    else:
-        return pairwise_probabilities
+    return [
+        (sum(team_prob) / denom) for team_prob in itertools.zip_longest(*[iter(pairwise_probabilities)] * (n-1))
```
Comment (philihp) [inline openskill/rate.py:221]:
```python
itertools.zip_longest(*[iter(pairwise_probabilities)] * (n-1))
```

this chunks the `pairwise_probabilities` list in n-1 chunks, which comes from https://stackoverflow.com/questions/312443/how-do-you-split-a-list-into-evenly-sized-chunks

Comment (philihp) [reply openskill/rate.py:221]: (i'm not very good at python, but i didn't know how to better do this)

Comment (daegontaven) [reply openskill/rate.py:221]: Looks good to me!

**Review by daegontaven — CHANGES_REQUESTED** [review]
Please format the code with `black .`  so that the checks can pass. Can you also add two spaces between the test functions for predicting wins as per [PEP 8 E302](https://www.flake8rules.com/rules/E302.html).

**Comment by philihp** [comment]
Sure thing! (sorry Python is a new language to me, I'll see what I can do)

**Comment by philihp** [comment]
@daegontaven should be good now

**Review by daegontaven — APPROVED** [review]

**Pull request merged**